scipy/spatial/transform/rotation.py

```diff
@@ -1463,6 +1464,73 @@ def magnitude(self):
         else:
             return angles
 
+    def reduce(self, left=None, right=None, return_best=False):
+        """Reduce this rotation with the provided rotation groups.
+
+        Parameters
+        ----------
+        left : `Rotation` instance
+            Object containing the left rotation group.
+        right : `Rotation` instance
+            Object containing the right rotation group.
+        right : bool
+            Whether to return the best indices
+
+        Returns
+        -------
+        reduced : `Rotation` instance
+            Object containing reduced rotations.
+        best_indices: integer array
+            Best indices of `left` and `right`
+        """
+        if left is None and right is None:
+            return cls(self._quat, normalized=True, copy=True)
+        elif right is None:
+            right = Rotation.from_quat([[0, 0, 0, 1.]])
+        elif left is None:
+            left = Rotation.from_quat([[0, 0, 0, 1.]])
+
+        p = self
+        lq = left.as_quat()
```

Comment:
Could you try to use more verbose names, maybe `left_quat, left_scalar, left_vector, self_quat, ...`. 

Basically use your best judgement to make code neater and more readable.

Reply:
I like the short notation.  To improve readability I have added a comment to describe the calculation which also explains the names.  I have also added a helper function, `split_rotation`, which simplifies the code.